platform/mbed_enum_flags.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2020 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_SCOPED_ENUM_FLAGS_H
+#define MBED_SCOPED_ENUM_FLAGS_H
+
+#ifdef __cplusplus
+#include <type_traits>
+
+#define ENUM_FLAG_OPERATOR(T,OP) \
+inline constexpr T operator OP( T lhs, T rhs) {\
```

Comment:
I'd put at least one space before `\` everywhere, and maybe consider doing block-style like mbed_atomic_impl.h.

Reply:
I will put in the spaces.

I will add the blocks